CONTRIBUTING.md

```diff
@@ -136,13 +136,34 @@ We follow [youtube-dl's policy](https://github.com/ytdl-org/youtube-dl#can-you-a
 
 Most users do not need to build yt-dlp and can [download the builds](https://github.com/yt-dlp/yt-dlp/releases) or get them via [the other installation methods](README.md#installation).
 
-To run yt-dlp as a developer, you don't need to build anything either. Simply execute
+To quickly run yt-dlp as a developer, you don't need to build anything either. Simply execute
 
-    python -m yt_dlp
+```shell
+$ python -m yt_dlp
+```
+
+To properly interact with `yt-dlp` as a developer, you should use [`hatch`](<https://hatch.pypa.io>) (easily installed using [`pipx`](<https://pipx.pypa.io>)).
+After installing, you can use `hatch shell` to create a shell with all dependencies as well as `yt-dlp` installed.
+
+If you plan on contributing to `yt-dlp`, run
+
+```shell
+$ hatch run install
```

Comment:
the only thing this does is install `pre-commit` (for now), right? IMO this part is a bit confusing

Reply:
Yes but it should be installed when contributing. We might potentially add more things later (though i doubt it)

Reply:
Oh and it creates the environment in the first place, installing the dependencies, and making subsequent calls a lot faster

Reply:
I dont think clarifying it as optional will help the cause. That just means people will not have to spend the time checkig it themselves and we will have to explain it for them.

I think explaining what the pre-commit hook does is the better approach. I tried to give a hint using the ` (run  hatch run fix  if it fails)` addon but not sure if and to what extend we should explain it in the contribution guidelines

Reply:
I think the new docs changes explain pre-commit in enough detail